packages/wonder-blocks-button/components/button-core.js

```diff
@@ -26,9 +26,9 @@ type Props = {|
     pressed: boolean,
 |};
 
-const StyledAnchor = addStyle("a");
-const StyledButton = addStyle("button");
-const StyledLink = addStyle(Link);
+const StyledAnchor = addStyle<"a">("a");
+const StyledButton = addStyle<"button">("button");
+const StyledLink = addStyle<typeof Link>(Link);
```

Comment:
Interesting that this can no longer be inferred. Makes it all a little less desirable. In most templating situations, either the argument is used to infer the template parameter, or the template parameter removes the need to specify the argument. This just seems super awkward - like the worst of all options.

Still, the code works. It's not your fault that the pattern isn't great, just musing out loud.

Reply:
I wish this was better too.